### ISSUE TITLE
Add a TimeSeries.add_data method

### ISSUE THREAD

**Issue by felixchenier** [kineticstoolkit/kineticstoolkit#200]
Currently, we already have `rename_data` and `remove_data`. But we don't have a `add_data` method. It was thought to be useless first, because we simply have to add something to the `data` dictionary. But since `data` must contain only NumPy arrays, it would be practical to use a `add_data` method to ensure that this is what we get.

This would have several advantages:

1. We would be always sure that the structure of the TimeSeries is correct (using a method is always safer than assigning anything to a dictionary).
2. This method could populate the time attribute if unset, so that it matches the length of the data.
3. This method could raise an error or at least warn if we add new data that mismatches the current data length.
4. This new method could accept any ArrayLike variable, and convert it to the desired NumPy array. This would be practical since we could add a list, a Series or a DataFrame directly and be sure to respect the structure of the TimeSeries.
5. For DataFrames, this method could check not only the length of the table, but also the index to be sure that the added data matches the time attribute of the TimeSeries, and raise an error or warn if it doesn't.

The signature would be something like:

```
def add_data(self, name : str, data : np.ArrayLike, *, overwrite : bool=??, in_place : bool=False)
```

**Comment by felixchenier** [comment]
I still think this method has sense just for sake of completeness since we already have `rename_data` and `remove_data`, and to be similar to `add_data_info`, `rename_data_info`, `remove_data_info`.

But two things are wrong with it:
1. It should not modify `time`. Side-effects are bad. The method to create `time` at the same time that `data` is probably at object creation: `ts = ktk.TimeSeries(array)` or ts = `ktk.TimeSeries(array, data_key="data")`. Then we get a full TimeSeries with both `time` and `data`.
2. These tests should be included as a property instead: See #206 and #207. The method `add_data` would differ to the `data` property only in terms of overwriting existing data.

**Comment by felixchenier** [comment]
On point 2: I finally decided against doing the tests on properties setters (see #207) since it added lots of friction for the user by disallowing any temporary operation and by adding cognitive load on something that should be simple: time is an array, and data is a dict of arrays. Period.
